factory/gen.go

```diff
@@ -244,7 +239,7 @@ func populateCN(secret *v1.Secret, cn ...string) *v1.Secret {
 		secret.Annotations = map[string]string{}
 	}
 	for _, cn := range cn {
-		if cnRegexp.MatchString(cn) {
+		if validHostnamePattern(cn) {
```

Comment:
I'm not sure this is correct, as populateCN is also called to add CNs extracted from TLS handshakes and HTTP host headers. Since `validHostnamePattern` calls `validHostname(host, true)`, I believe this would allow unauthenticated callers to make dynamiclistener add wildcard SANs, if they spoof the correct value. I think wildcards should only be added if requested by the administrator.